velox/expression/Expr.cpp

```diff
@@ -1207,8 +1262,26 @@ void Expr::evalAll(
   LocalSelectivityVector mutableRemainingRowsHolder(context);
 
   inputValues_.resize(inputs_.size());
+  EvalCtx::ErrorVectorPtr argumentErrors;
+  EvalCtx::ErrorVectorPtr errors;
+  if (context.errors()) {
+    context.swapErrors(errors);
+  }
+
   for (int32_t i = 0; i < inputs_.size(); ++i) {
+    ScopedVarSetter throwErrors(
+        context.mutableThrowOnError(), throwArgumentErrors);
     inputs_[i]->eval(*remainingRows, context, inputValues_[i]);
+    if (context.errors()) {
+      // New errors were produced, add them to argumentErrors.
+      if (argumentErrors) {
```

Comment:
shall we put this if{}else{} in a single function. 
maybe pass context to union errors